Add API endpoint on eCR Viewer to save FHIR bundle

angelathe opened the pull request:
# PULL REQUEST

## Summary
Create an API endpoint on the eCR Viewer container that will save a FHIR bundle to Postgres. Request body takes in a `fhirBundle` and a `saveSource` -- see docstring for more info.

## Related Issue
Fixes #1331 

## Additional Information: background on why we didn't include tests
- Tried writing tests with Jest, but came into a host of issues (Related Link here: ([Async Server Components])(https://nextjs.org/docs/app/building-your-application/testing#async-server-components)
- Next, tried E2E testing, where I wrote tests in Cypress that stubbed the API response. However, the tests were just asserting directly on our own mocked response, and we couldn't (yet) figure out a way to test the outgoing db request.

## Checklist
- [ ] Make a POST request to http://localhost:3000/api/save-fhir-data 
- [ ] Example request body below:
`{
    "fhirBundle": {
        "resourceType": "Bundle",
        "type": "batch",
        "entry": [
        {
            "fullUrl": "urn:uuid:12345",
            "resource": {
                "resourceType": "Composition",
                "id": "12345"
            }
        }
        ]
    },
    "saveSource": "postgres"
}`

JNygaard-Skylight commented on containers/ecr-viewer/src/app/api/save-fhir-data/route.ts:
```diff
@@ -0,0 +1,86 @@
+
+      try {
+        const saveECR = await database.one(addFhir);
+        console.log("Created Resource with: ", saveECR);
```
You should remove this `console.log`

angelathe replied:
done!

JNygaard-Skylight reviewed (approved):
Looks good to me!

nickbristow reviewed (changes requested):
because this will also save to s3, we should change the route from `save-fhir-data` to something like `save-fhir-data-db` or something like that, im bad at names

angelathe commented:
> because this will also save to s3, we should change the route from `save-fhir-data` to something like `save-fhir-data-db` or something like that, im bad at names

@nickbristow My impression was that the next ticket (#1344) updates this endpoint to also include s3 (rather than making a new endpoint)? If so, not sure if `db` would make sense to have in the route name

angelathe commented:
> > because this will also save to s3, we should change the route from `save-fhir-data` to something like `save-fhir-data-db` or something like that, im bad at names
> 
> @nickbristow My impression was that the next ticket (#1344) updates this endpoint to also include s3 (rather than making a new endpoint)? If so, not sure if `db` would make sense to have in the route name

Note: Added `saveSource` property in request body that indicates where the FHIR data should be saved (`postgres` or `s3`). Docstrings are also updated!
